### PR TITLE
KOA-4619 Update BpkHorizontalNav to match Figma

### PR DESCRIPTION
**Pull request by marianeum**
| Before  | After |
| ------------- | ------------- |
| default_b0bf34c031f9bbb16a9beaa8aac11996  | default  |
| default_dm_666e5f4d2216dccf60cf0f98aa5c5f1c  | default_dm  |

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/src/main/java/net/skyscanner/backpack/horisontalnav/BpkHorizontalNav.kt [hunk 1]
```diff
@@ -223,9 +217,7 @@ open class BpkHorizontalNav @JvmOverloads constructor(
         it.setBackground(ColorStateList.valueOf(Color.TRANSPARENT), tabTextColors!!)
         it.setTextColor(tabTextColors)
       }
-      findViewById<TextView>(android.R.id.text1)?.let {
-        it.setTextColor(tabTextColors)
-      }
+      findViewById<TextView>(android.R.id.text1)?.setTextColor(tabTextColors)
```
Comment (marianeum) [inline Backpack/src/main/java/net/skyscanner/backpack/horisontalnav/BpkHorizontalNav.kt:220]: just cleaned up a warning - the let isn't needed as we're just setting one property

File: Backpack/src/main/java/net/skyscanner/backpack/horisontalnav/BpkHorizontalNav.kt [hunk 2]
```diff
@@ -110,8 +108,6 @@ open class BpkHorizontalNav @JvmOverloads constructor(
   private val texts = SparseArrayCompat<CharSequence?>()
   private val notificationDots = SparseBooleanArray()
   private val badges = SparseArray<CharSequence?>()
-  @SuppressLint("RestrictedApi")
-  private val allCaps = AllCapsTransformationMethod(context)
```
Comment (marianeum) [inline Backpack/src/main/java/net/skyscanner/backpack/horisontalnav/BpkHorizontalNav.kt:114]: no more all caps 🎉

File: Backpack/src/main/java/net/skyscanner/backpack/horisontalnav/BpkHorizontalNav.kt [hunk 3]
```diff
@@ -59,7 +57,7 @@ open class BpkHorizontalNav @JvmOverloads constructor(
     Normal(
       id = 0,
       styleAttribute = R.attr.bpkHorizontalNavStyle,
-      defaultTextColor = R.color.__horizontalNavTextDefault,
+      defaultTextColor = R.color.bpkTextPrimary,
```
Comment (marianeum) [inline Backpack/src/main/java/net/skyscanner/backpack/horisontalnav/BpkHorizontalNav.kt:60]: one custom font colour down 🎉

File: app/src/androidTest/java/net/skyscanner/backpack/horizontalnav/BpkHorizontalNavTest.kt [hunk 4]
```diff
@@ -28,7 +28,7 @@ import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
-class HorizontalNavTests : BpkSnapshotTest() {
+class BpkHorizontalNavTest : BpkSnapshotTest() {
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/horizontalnav/BpkHorizontalNavTest.kt:31]: renamed this to match the conventions

File: app/src/androidTest/java/net/skyscanner/backpack/horizontalnav/BpkHorizontalNavTest.kt [hunk 5]
```diff
@@ -83,8 +83,8 @@ class HorizontalNavTests : BpkSnapshotTest() {
   }
 
   private fun BpkHorizontalNav.init() = apply {
-    addTab(newTab().setText("1"))
-    addTab(newTab().setText("2"))
-    addTab(newTab().setText("3"))
+    addTab(newTab().setText("Tab 1"))
```
Comment (marianeum) [inline app/src/androidTest/java/net/skyscanner/backpack/horizontalnav/BpkHorizontalNavTest.kt:86]: added a text so it actually detects all caps vs not

**Review by olliecurtis — APPROVED** [review]
LGTM!

**Pull request merged**